crates/consensus/auto-seal/src/mode.rs

```diff
@@ -100,10 +100,25 @@ impl FixedBlockTimeMiner {
     where
         Pool: TransactionPool,
     {
+        // Prevents tx spams that breaks instant miner
         if self.interval.poll_tick(cx).is_ready() {
-            // drain the pool
-            return Poll::Ready(pool.best_transactions().collect())
+            let mut txs = Vec::new();
+            let mut cumulative_gas = 0;
+
+            for tx in pool.best_transactions() {
+                let next_gas = cumulative_gas + tx.transaction.gas_limit();
+
+                if next_gas > ETHEREUM_BLOCK_GAS_LIMIT {
+                    break
+                }
```

Comment:
you should try to fit any remaining transaction instead of breaking from the loop here